### PR TITLE
In Debian 12 the which command has been removed to submit patches.

### PR DESCRIPTION
**Pull request by search5**
Patch function execution error caused by the deletion of the which command in Debian 12.

Currently, only Debian 12 has removed the which command, but I suspect it will be removed in many future distributions as well.

Accordingly, we submit a patch that applies commands that can be used equally in many operating systems as well as Debian.

Please apply the patch.

### CODE REVIEW

**Comment by qparis** [comment]
Thank you for your patch. Is command -v available on all distribution?

**Comment by search5** [comment]
Yes, command is not an executable file according to the posix standard and
is implemented in all shells conforming to the posix standard.

2023년 8월 12일 (토) 오전 8:15, Quentin Pâris ***@***.***>님이 작성:

**Comment by qparis** [comment]
Ok thank you. 
Then I'll merge the patch

File: python/mainwindow.py [hunk 1]
```diff
@@ -1216,7 +1217,8 @@ def OnInit(self):
     def _executableFound(self, executable):
         devnull = open('/dev/null', 'wb')
         try:
-            returncode = subprocess.call(["which", executable], stdout=devnull)
+            executable_path = shutil.which(executable)
+            returncode = subprocess.call(executable_path, stdout=devnull)
```
Comment (robert-scheck) [inline python/mainwindow.py:1221]: I am sorry, but this is incorrect: `returncode` gets the exit code from calling e.g. `/usr/bin/nc` assigned, which is in my case `2` (when using `ncat` from nmap, the default at various Linux distributions, such as Fedora). This return code says nothing whether the desired executable exists or whether it's usable.

Comment (qparis) [reply python/mainwindow.py:1221]: You are correct. We should use the expression `shutil.which(executalbe) is None` instead

Comment (qparis) [reply python/mainwindow.py:1221]: (or is not None)

Comment (robert-scheck) [reply python/mainwindow.py:1221]: I've proposed a fix with #74, just let me know if you prefer another logic or code style or so.

**Pull request merged**